extensions/oidc/runtime/src/main/java/io/quarkus/oidc/AuthenticationContext.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.oidc;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation which can be used to enforce required authentication level for the endpoint classes and methods.
+ * <a href="https://datatracker.ietf.org/doc/rfc9470/">OAuth 2.0 Step Up Authentication Challenge Protocol</a>
+ * is initiated when the access token does not have sufficient authentication strength.
```

Comment:
We may need to tune this message a bit since we allow it against ID token. I'll suggest something once it is ready to go otherwise, thanks

Reply:
I addressed all the comments, moved tests and added one more. Waiting for more feedback.